### ISSUE TITLE
Display emoji

### ISSUE THREAD

**Issue by tisnatch** [heskja/MMM-CalendarWeek#30]
As an example, Google Calendar allows you to display emoji in event titles. The emoji are not imported correctly into the MM interface. Would it be possible to correctly show emoji?

**Comment by heskja** [comment]
Thank you for reporting this. I'll investigate this one and get back to you.

**Comment by heskja** [comment]
Hi,
I have looked into this, and I am unable to reproduce this. I added an event with the emoji "🌸", and it displays correctly. Verified using Safari on OS X. 

Can you please provide more details on your environment and the emojis in question?

**Comment by secastles** [comment]
Will do. But.... I need to do it about 10 days from now. Will be OOO for a
while.

**Comment by secastles** [comment]
Hi, sorry for the late reply. OK, just tested again with a few events.

Environment:
- I _think_ I'm on Electron 2.0.1 (found this after calling "npm i electron-version -S"... bear with me, I'm a bit of a noob in Linux)
- I'm importing events from google calendar
- To add an emoji I copy/pasted from here "https://getemoji.com/"
- I copy pasted different emoji to the title of a few events
- I tested by adding the emoji as the first character of the event name
- Emoji display corrected in Chrome in google calendar
- Emoji don't display in MM. Instead there is a placeholder box/rectangle/square character

Please let me know how I can help troubleshoot

**Comment by secastles** [comment]
Just updated my MM install today. Basically git clone then npm install. Updates went fine and I'm back up and running. But no emoji.  Also made sure to update raspbian as well.

**Comment by heskja** [comment]
I found this issue on the original MagicMirror repository; https://github.com/MichMich/MagicMirror/issues/1687

Seems to be an issue with displaying emojis on the Pi in general. Have you tried adding more "emoji-friendly" fonts yo your Pi OS?

**Comment by secastles** [comment]
I'll dig into how to do that. But help me reason out the chain of how this is going to work: (1) install "emoji friendly fonts" in raspbian (2) presumably electron then gets automatic access to those (3) magicmirror automatically displays emoji -or- i need to change the font used in magicmirror to an emoji friendly one?

**Comment by secastles** [comment]
I did some googling. doesn't really seem like a way to get emoji to display in the raspbian gui (e.g. in a browser). There's a hack from three years ago that, seemly, ends up breaking a bunch of other stuff. I couldn't find dick about emoji support in electron outside of a custom library for node.js. Sooo.. guess that just means it's not supported? Which brings me to: how did you get emoji to show up on a Pi? You mentioned Safari on OSX but... that's not a Pi or MagicMirror.

**Comment by heskja** [comment]
Hi,
I am using a MagicMirror in server-only mode on a virtual machine for my development. This way I can easily snapshot/backup/restore the enivornmenx if I mess stuff up, so therefore I tested using OSX. 

This means, as far as I can tell, that Electron will display the emojis correctly without any custom libraries for node.js. I am not sure what kind of fonts you would have to install, but some quick searches on google yields multiple results on how to install fonts on Raspbian. Example:
https://raspberrytips.com/install-fonts-raspberry-pi/
https://raspberry-projects.com/pi/programming-in-c/gui-programming-in-c/gtk/text/fonts

You could also try the MacicMirror forum https://forum.magicmirror.builders for tips. 

I will close this issue as there is nothing I can do to my module to "fix" this, as nothing is broken.

**Comment by secastles** [comment]
Your googlefu is better than mine. I installed Google's Noto emoji fonts following this process:
`sudo cp *ttf /usr/share/fonts/truetype/noto`
`sudo fc-cache -f -v`

Then I reboot. No dice. Still won't display emoji. Yeah, it's not a bug for MagicMirror. It's something else with raspbian or maybe with the way electron works with raspbian.

**Comment by secastles** [comment]
I just tried something new. I installed the Symbola font and now I can see black and white emoji! I don't know anything about fonts but I assume raspbian is translating the color emoji to the closest thing it can find in the Symbola font. It does a decent enough job.  I think this post might take it the rest of the way but it appears to have some side effects that might not be desirable: https://www.omgubuntu.co.uk/2016/08/enable-color-emoji-linux-google-chrome-noto

Just posting this for anyone who comes after me looking for info on fonts/MM/rasberrypi
